cmake/EnableSanitizers.cmake

```diff
@@ -38,44 +38,64 @@ option(SANITIZE_ADDRESS "Enable AddressSanitizer for the build." "")
 if (SANITIZE_ADDRESS)
     set(ASAN_CANDIDATES "-fsanitize=address -fno-omit-frame-pointer" "-fsanitize=address")
     sanitizer_test(AddressSanitizer ${ASAN_CANDIDATES})
+    if (NOT AddressSanitizer_ENABLED)
+        message(FATAL_ERROR "AddressSanitizer requested but could not be enabled.")
+    else()
+        message(STATUS "AddressSanitizer is enabled.")
+    endif ()
 endif (SANITIZE_ADDRESS)
 
 option(SANITIZE_LEAKS "Enable LeakSanitizer for the build." "")
 if (SANITIZE_LEAKS)
     set(ASAN_CANDIDATES "-fsanitize=leaks -fno-omit-frame-pointer" "-fsanitize=leaks")
     sanitizer_test(LeaksSanitizer ${ASAN_CANDIDATES})
+    if (NOT LeaksSanitizer_ENABLED)
+        message(FATAL_ERROR "LeaksSanitizer requested but could not be enabled.")
+    else()
+        message(STATUS "LeaksSanitizer is enabled.")
+    endif ()
 endif (SANITIZE_LEAKS)
 
 option(SANITIZE_MEMORY "Enable MemorySanitizer for the build." "")
 if (SANITIZE_MEMORY)
-    if (${AddressSanitizer_ENABLED})
+    if (AddressSanitizer_ENABLED)
         message(FATAL_ERROR "Cannot enable MemorySanitizer when AddressSanitizer is enabled")
     else()
-        set(MSAN_CANDIDATES "-fsanitize=memory -fno-omit-frame-pointer" "-fsanitize=memory")
+        set(MSAN_CANDIDATES
+	  "-fsanitize=memory -fno-omit-frame-pointer -fsanitize-memory-track-origins=2"
```

Comment:
What does it mean to have 3 sets of flags? Does it run in 3 modes? Does it pick one? Do you pick one?

Reply:
The first one that works.  Added a comment.